src/bennu/devices/modules/comms/iec60870-5/module/Server.cpp

```diff
@@ -282,6 +404,95 @@ bool Server::interrogationHandler(void *parameter, IMasterConnection connection,
     return true;
 }
 
+/*
+* Callback handler for interrogation messages that reports indications as double point values
+*/
+bool Server::interrogationHandlerDoublePoint(void *parameter, IMasterConnection connection, CS101_ASDU asdu, uint8_t qoi)
+{
+    printf("Received interrogation for group %i\n", qoi);
+
+    if (qoi == 20) /* only handle station interrogation */
+    {
+        CS101_AppLayerParameters alParams = IMasterConnection_getApplicationLayerParameters(connection);
+        IMasterConnection_sendACT_CON(connection, asdu, false);
+
+        // Send binary values
+        CS101_ASDU newAsdu = CS101_ASDU_create(alParams, false, CS101_COT_INTERROGATED_BY_STATION, 0, 1, false, false);
+        // kv = {<address>: {<tag>, eInput}}
+        for (const auto &kv : gServer->mBinaryPoints)
+        {
+            const std::string tag = kv.second.first;
+            if (CS101_ASDU_getPayloadSize(newAsdu) < MAX_ASDU_PAYLOAD_SIZE)
```

Comment:
See earlier comment above about duplicated logic and inverting the if statement

Reply:
fixed